build_runner/lib/src/daemon/daemon_builder.dart

```diff
@@ -106,6 +109,22 @@ class BuildRunnerDaemonBuilder implements DaemonBuilder {
               verbose: _buildOptions.verbose)
           .toString()));
 
+  void _signalEnd(Iterable<daemon.BuildResult> results) {
+    _buildingCompleter.complete();
+    _buildResults.add(daemon.BuildResults((b) => b..results.addAll(results)));
+  }
+
+  void _signalStart(Iterable<String> targets) {
+    if (_buildingCompleter.isCompleted) _buildingCompleter = Completer();
```

Comment:
You are correct, this was done to specifically handle the initial build case.

I guess it makes more sense to handle the initial build scenario in webdev instead of here so I modified the code per your suggestion. However, the assert and explicitly setting to null is not necessary imo. The daemon ensures that at most one build is initiated at a time.